blocks/library/more/index.js

```diff
@@ -68,33 +68,54 @@ export const settings = {
 		],
 	},
 
-	edit( { attributes, setAttributes, isSelected } ) {
-		const { customText, noTeaser } = attributes;
-
-		const toggleNoTeaser = () => setAttributes( { noTeaser: ! noTeaser } );
-		const defaultText = __( 'Read more' );
-		const value = customText !== undefined ? customText : defaultText;
-		const inputLength = value.length ? value.length + 1 : 1;
-
-		return [
-			isSelected && (
-				<InspectorControls key="inspector">
-					<ToggleControl
-						label={ __( 'Hide the teaser before the "More" tag' ) }
-						checked={ !! noTeaser }
-						onChange={ toggleNoTeaser }
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.onChangeInput = this.onChangeInput.bind( this );
+
+			this.state = {
+				defaultText: __( 'Read more' ),
+			};
+		}
+
+		onChangeInput( event ) {
+			this.setState( {
+				defaultText: '',
+			} );
+
+			const value = event.target.value.length === 0 ? undefined : event.target.value;
+			this.props.setAttributes( { customText: value } );
+		}
+
+		render() {
+			const { customText, noTeaser } = this.props.attributes;
+			const { setAttributes, isSelected } = this.props;
+
+			const toggleNoTeaser = () => setAttributes( { noTeaser: ! noTeaser } );
+			const defaultText = this.state.defaultText;
+			const value = customText !== undefined ? customText : defaultText;
+			const inputLength = value.length ? value.length + 1 : 1;
```

Comment:
I know this existed previously, but it caught my eye: In what case would `const inputLength = value.length + 1;` be insufficient? If length is zero, `0 + 1` is still `1`, so the else of the ternary is redundant.